src/hist/intervals.py

```diff
@@ -73,7 +74,7 @@ def poisson_interval(
     interval_min = scale * stats.chi2.ppf((1 - coverage) / 2, 2 * counts) / 2.0
     interval_max = scale * stats.chi2.ppf((1 + coverage) / 2, 2 * (counts + 1)) / 2.0
     interval = np.stack((interval_min, interval_max))
-    interval[interval == np.nan] = 0.0  # chi2.ppf produces nan for counts=0
+    interval[np.isnan(interval)] = 0.0  # chi2.ppf produces nan for counts=0
```

Comment:
Optional:

```suggestion
    interval = np.nan_to_num(interval, copy=False)  # chi2.ppf produces nan for counts=0
```

This also replaces infinity with very large values, but I think this is safe here.

Reply:
I realize this is preventing nan propagation in the inputs, so I am changing to `interval[values == 0.0] = 0.0` to treat the specific case only.